python/src/spark_rapids_ml/metrics/RegressionMetrics.py

```diff
@@ -0,0 +1,252 @@
+        self._curr_weight_sum = [total_cnt] * self._num_cols
+
+    def merge(self, other: "_SummarizerBuffer") -> "_SummarizerBuffer":
+        """Merge the other into self and return a new SummarizerBuffer"""
+        self._total_cnt += other._total_cnt
+        self._total_weight_sum += other._total_weight_sum
+        self._weight_square_sum += other._weight_square_sum
+
+        for i in range(self._num_cols):
+            this_weight_sum = self._curr_weight_sum[i]
+            other_weight_sum = other._curr_weight_sum[i]
+            total_weight_sum = this_weight_sum + other_weight_sum
+
+            if total_weight_sum != 0.0:
+                delta_mean = other._curr_mean[i] - self._curr_mean[i]
+                # merge mean together
+                self._curr_mean[i] += delta_mean * other_weight_sum / total_weight_sum
+                # merge m2n together
+                self._curr_m2n[i] += (
+                    other._curr_m2n[i]
+                    + delta_mean
+                    * delta_mean
+                    * this_weight_sum
+                    * other_weight_sum
+                    / total_weight_sum
+                )
+
+            self._curr_weight_sum[i] = total_weight_sum
+            self._curr_m2[i] += other._curr_m2[i]
+            self._curr_l1[i] += other._curr_l1[i]
+
+        return _SummarizerBuffer(
+            self._curr_mean,
+            self._curr_m2n,
+            self._curr_m2,
+            self._curr_l1,
+            self._total_cnt,
+        )
+
+    @property
+    def total_count(self) -> int:
+        return self._total_cnt
+
+    @property
+    def m2(self) -> List[float]:
+        """\sum_{i=1}^{N}(x_i)^{2} of each dimension"""
+        return self._curr_m2
+
+    @property
+    def norm_l2(self) -> List[float]:
+        """L2 (Euclidean) norm of each dimension."""
+        real_magnitude = [math.sqrt(m2) for m2 in self._curr_m2]
+        return real_magnitude
+
+    @property
+    def norm_l1(self) -> List[float]:
+        """L1 norm of each dimension."""
+        return self._curr_l1
+
+    @property
+    def mean(self) -> List[float]:
+        """mean of each dimension."""
+        real_mean = [
+            self._curr_mean[i] * (self._curr_weight_sum[i] / self._total_weight_sum)
+            for i in range(self._num_cols)
+        ]
+        return real_mean
+
+    def _compute_variance(self) -> List[float]:
+        denominator = self._total_weight_sum - (
+            self._weight_square_sum / self._total_weight_sum
+        )
+        if denominator > 0.0:
+            delta_mean = self._curr_mean
+            real_variance = [
+                max(
+                    (
+                        self._curr_m2n[i]
+                        # + delta_mean[i]
+                        # * delta_mean[i]
+                        # * self._curr_weight_sum[i]
+                        # * (self._total_weight_sum - self._curr_weight_sum[i])
+                        # / self._total_weight_sum
```

Comment:
Why leave this commented out code?

Reply:
Ok, I will delete them in the following PR.